adapters/repos/db/migrator_test.go

```diff
@@ -0,0 +1,372 @@
+		class1 = &models.Class{
+			Class:               "C1",
+			InvertedIndexConfig: &models.InvertedIndexConfig{},
+			VectorIndexConfig:   hnsw.UserConfig{Distance: vectorIndex.DefaultDistanceMetric},
+		}
+
+		invertedIdxConfig = &models.InvertedIndexConfig{
+			CleanupIntervalSeconds: 60,
+			Stopwords: &models.StopwordConfig{
+				Preset: "none",
+			},
+			IndexNullState:      true,
+			IndexPropertyLength: true,
+		}
+	)
+
+	err := mi.UpdateInvertedIndexConfig(ctx, "NonExistingClass", &models.InvertedIndexConfig{})
+	r.NotNil(err)
+
+	err = mi.AddClass(ctx, class1, &sharding.State{})
+	r.Nil(err)
+
+	err = mi.UpdateInvertedIndexConfig(ctx, class1.Class, invertedIdxConfig)
+	r.Nil(err)
+	r.True(mi.db.indices[indexID("C1")].getInvertedIndexConfig().IndexNullState)
+
+	invertedIdxConfig.IndexNullState = false
+	err = mi.UpdateInvertedIndexConfig(ctx, class1.Class, invertedIdxConfig)
+	r.Nil(err)
+	r.False(mi.db.indices[indexID("C1")].getInvertedIndexConfig().IndexNullState)
+}
+
+func TestUpdateVectorIndexConfig(t *testing.T) {
+	var (
+		ctx    = context.Background()
+		r      = require.New(t)
+		mi     = newMigrator(t, 1)
+		class1 = &models.Class{
+			Class:               "C1",
+			InvertedIndexConfig: &models.InvertedIndexConfig{},
+			VectorIndexConfig:   hnsw.UserConfig{Distance: vectorIndex.DefaultDistanceMetric},
+		}
+	)
+
+	err := mi.UpdateVectorIndexConfig(ctx, "NonExistingClass", hnsw.UserConfig{Distance: vectorIndex.DefaultDistanceMetric})
+	r.NotNil(err)
+
+	err = mi.AddClass(ctx, class1, &sharding.State{})
+	r.Nil(err)
+
+	err = mi.UpdateVectorIndexConfig(ctx, "C1", hnsw.UserConfig{Distance: vectorIndex.DistanceHamming})
+	r.Nil(err)
+	r.Equal(vectorIndex.DistanceHamming, mi.db.indices[indexID("C1")].getVectorIndexConfig().DistanceName())
+
+	err = mi.UpdateVectorIndexConfig(ctx, "C1", hnsw.UserConfig{Distance: vectorIndex.DistanceManhattan})
+	r.Nil(err)
+	r.Equal(vectorIndex.DistanceManhattan, mi.db.indices[indexID("C1")].getVectorIndexConfig().DistanceName())
+
+	// confirm idempotence
+	err = mi.UpdateVectorIndexConfig(ctx, "C1", hnsw.UserConfig{Distance: vectorIndex.DistanceManhattan})
+	r.Nil(err)
+	r.Equal(vectorIndex.DistanceManhattan, mi.db.indices[indexID("C1")].getVectorIndexConfig().DistanceName())
+}
+
+func newMigrator(t *testing.T, numClasses int, classes ...*models.Class) *Migrator {
+	states := make(map[string]*sharding.State, numClasses)
+	err := os.MkdirAll(t.TempDir(), os.ModePerm)
+	require.Nil(t, err)
+	db := testDB(t.TempDir(), classes, states)
+	db.indices = make(map[string]*Index)
+	return NewMigrator(db, db.logger)
+}
+
+func classIsInMemoryAndFileSystem(r *require.Assertions, mi *Migrator, c *models.Class) {
+	existedIndex := mi.db.GetIndex(schema.ClassName(c.Class))
+	r.Equal(existedIndex.getVectorIndexConfig(), c.VectorIndexConfig)
+	existedClasses := existedIndex.getSchema.GetSchemaSkipAuth().Objects.Classes
+	r.Equal(existedClasses[0], c)
+	_, err := os.Stat(existedIndex.path())
+	r.Nil(err)
```

Comment:
I believe just checking if a class exits is not sufficient. We need to validate that all side effects (e.g shards) are present in filesystem and all those that should not be there are absent.

Reply:
i've added a new test case to check if the `state.Physical` change would reflect in the file system, lets see what DB team suggestions would be